pkg/cri/resource-manager/control/blockio/blockio.go

```diff
@@ -133,25 +164,282 @@ func (ctl *blockio) BlockIOClass(c cache.Container) string {
 	if cclass == "" {
 		cclass = string(c.GetQOSClass())
 	}
-
-	bioclass, ok := opt.Classes[cclass]
+	_, ok := opt.Classes[cclass]
 	if !ok {
-		if bioclass, ok = opt.Classes["*"]; !ok {
-			bioclass = cclass
+		if _, ok := opt.Classes["*"]; !ok {
+			return ""
+		} else {
+			return "*"
```

Comment:
Hmm, what is the logic/reasoning here ? Do we want to support a verbatim class name of "*" ? The original code was using that as a fallback/wildcard for class name mapping like this:
  - start from block I/O class set for the container, falling back to QoS class is none is set
  - if the class is (explicitly) mapped to something use that, otherwise
  - if the class is wildcard-mapped to something use that, otherwise
  - use the class set or derived from QoS class

Reply:
My initial thought with the "*" class was to support setting weights/throttling for all pods that do not belong to any other BlockIO or QOS class that is mentioned in the configuration.

The benefit would be that the same configuration would not need to be repeated. For example, one could set lower weight for Burstable and BestEffort classes by using class "*" and then define only Guaranteed class to assign higher weights.

The big drawback would be that the class configuration logic would then differ from what is already in RDT.

What do you think, would it be good to support a "default" class in configuration(s)? Or will I take this back to the original logic (copy from RDT), and we can possibly reconsider class matching for all control blocks later, if needed?

Reply:
Yes. Defaults definitely make sense. That's what '*' was meant for, but it was imagined to be used in a slightly different way, like in the RDT controller (IIRC). There the mapping happens in the the controller level while the classes are defined in low-level RDT implementation. So for instance you would define 'default' and 'Guaranteed' classes in the actual block I/O implementation (pkg/cgroups blockio) and map BestEffort and Burstable to 'default' in the controller.

I am not sure if this separation of the mapping from the class definitions really makes sense. It might be actually more confusing this way than if we had it in a single place. So I'm fine with either way (or anything else if it makes sense), but generally speaking I think we should try to keep the configuration approach to the RDT and block I/O controllers consistent unless there is a pressing reason to diverge.

Reply:
Agreed with next steps: move logic from control/blockio to pkg/blockio and create similar class mapping to control/blockio as there is in contorl/rdt